src/main/java/org/openstreetmap/atlas/checks/validation/areas/TallBuildingCheck.java

```diff
@@ -0,0 +1,728 @@
+    private final double bufferDistanceMeters;
+    private static final double MIN_DATASET_SIZE_DEFAULT = 50;
+    private final double minDatasetSizeForStatsComputation;
+    private final double maxLevelTagValue;
+    private static final double MAX_LEVEL_TAG_VALUE_DEFAULT = 100;
+    private static final double OUTLIER_MULTIPLIER_DEFAULT = 3;
+    private final double outlierMultiplier;
+    private final Map<Rectangle, Tuple4<String, Double, Double, Double>> storedAreasWithStatistics = new HashMap<>();
+    private final Set<String> invalidHeightCharacters;
+    private static final Set<String> INVALID_CHARACTER_DEFAULT = Set.of("~", "`", "!", "@", "#",
+            "$", "%", "^", "&", "*", "(", ")", "-", "_", "+", "=", "{", "[", "}", "]", "|", "\\",
+            ":", ";", "<", ",", ">", "?", "/");
+    private final double magicNumber3;
+    private static final double THREE_DEFAULT = 3.0;
+    private final double magicNumber4;
+    private static final double FOUR_DEFAULT = 4.0;
+    private final double magicNumberOneQuarter;
+    private static final double ONE_QUARTER_DEFAULT = 0.25;
+    private final double magicNumberThreeQuarters;
+    private static final double THREE_QUARTER_DEFAULT = 0.75;
+
+    /**
+     * The default constructor that must be supplied. The Atlas Checks framework will generate the
+     * checks with this constructor, supplying a configuration that can be used to adjust any
+     * parameters that the check uses during operation.
+     *
+     * @param configuration
+     *            the JSON configuration for this check
+     */
+    public TallBuildingCheck(final Configuration configuration)
+    {
+        super(configuration);
+        this.bufferDistanceMeters = this.configurationValue(configuration, "bufferDistanceMeters",
+                BUFFER_DISTANCE_DEFAULT);
+        this.minDatasetSizeForStatsComputation = this.configurationValue(configuration,
+                "minDatasetSizeForStatsComputation", MIN_DATASET_SIZE_DEFAULT);
+        this.maxLevelTagValue = this.configurationValue(configuration, "maxLevelTagValue",
+                MAX_LEVEL_TAG_VALUE_DEFAULT);
+        this.outlierMultiplier = this.configurationValue(configuration, "outlierMultiplier",
+                OUTLIER_MULTIPLIER_DEFAULT);
+        this.invalidHeightCharacters = new HashSet<>(this.configurationValue(configuration,
+                "invalidHeightCharacters", INVALID_CHARACTER_DEFAULT));
+        this.magicNumber3 = this.configurationValue(configuration, "magicNumbers.3", THREE_DEFAULT);
+        this.magicNumber4 = this.configurationValue(configuration, "magicNumbers.4", FOUR_DEFAULT);
+        this.magicNumberOneQuarter = this.configurationValue(configuration,
+                "magicNumbers.oneQuarter", ONE_QUARTER_DEFAULT);
+        this.magicNumberThreeQuarters = this.configurationValue(configuration,
+                "magicNumbers.threeQuarters", THREE_QUARTER_DEFAULT);
```

Comment:
Please refer to the naming convention used throughout the file for the config keys.